OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -271,7 +273,7 @@ protected override void OnFirstRun(Actor self)
 			if (target.IsValidFor(self))
 			{
 				QueueChild(new Fly(self, target, target.CenterPosition));
-				QueueChild(new Fly(self, target, exitRange, WDist.MaxValue, target.CenterPosition));
+				QueueChild(new FlyForward(self, exitRange));
```

Comment:
Thinking some more about it, `StrafeAttackRun` would need some adjustment as well for sliding aircraft. As I said above, the sliding aircraft will still be busy turning towards the target when it is already moving back towards the target. In order to account for that in the case when you explicitly want a strafe, we would need to add an extra distance on top of `exitRange` so that the aircraft is fully facing the target by the time it reaches the required range.

Something like
```C#
exitRange += aircraft.Info.Speed * 128 / aircraft.Info.TurnSpeed;
```
would do. We could of course require modders to just think of this themselves and set it manually, but this way it is actually more consistent with non-sliding aircraft which also do their turning outside of the `exitRange` and it also plays nice with the default value (the maximum weapon range).

I'm thinking that for aircraft that can slide _and_ can hover it would make more sense to just turn in place between strafing runs instead.

Reply:
Actually, something like this would be even better:
```C#
QueueChild(new Fly(self, target, target.CenterPosition));
QueueChild(new FlyForward(self, exitRange));
QueueChild(new Fly(self, target, exitRange + new WDist(aircraft.Info.Speed * 64 / aircraft.Info.TurnSpeed), WDist.MaxValue, target.CenterPosition));
```